frame/building-apps-sdk.md

```diff
@@ -0,0 +1,757 @@
+<summary>Python</summary>
+
+#### Python
+{: .no_toc }
+
+```python
+async def evaluate(self, lua_expression: str, timeout: Optional[float] = 10) -> str
+```
+
+Examples:
+
+```python
+print(await frame.evaluate("1+2"))
+# prints 3
+
+print(await frame.evaluate("frame.battery_level() > 50"))
+# prints True or False
+
+print(await frame.evaluate("'w00t'"))
+# prints w00t
+
+# will throw an exception if there is no value returned or if there is a syntax error
+print(await frame.evaluate("not_defined"))
+```
+
+</details>
+
+---
+
+## System Functions
+
+### Get Battery Level
+
+```python
+async frame.get_battery_level() -> int
+```
+
+Gets the Frame battery level as an integer from 0 to 100.  Equivalent to `await self.evaluate("frame.battery_level()")`.
+
+<details markdown="block">
+<summary>Python</summary>
+
+#### Python
+{: .no_toc }
+```python
+async def get_battery_level(self) -> int
+```
+
+Example:
+
+```python
+print(f"Frame battery: {await frame.get_battery_level()}%")
+```
+
+</details>
+
+
+### Sleep
+
+```python
+async frame.sleep(seconds: Optional[float])
+```
+
+Sleeps for a given number of seconds. If no argument is given, Frame will go to sleep until a tap gesture wakes it up.  This does not block, so your calling code will continue even while Frame sleeps.
+
+* `seconds` *(optional float)*: Seconds do sleep at a float, such as 1.25 or 0.01.  If not specified (or `None`/`null`), Frame will go to sleep until a tap gesture wakes it up.
+
+<details markdown="block">
+<summary>Python</summary>
+
+#### Python
+{: .no_toc }
+```python
+async def sleep(self, seconds: Optional[float])
+```
+
+Examples:
+
+```python
+# wait for 5 seconds
+await frame.sleep(5)
+print("Frame is sleeping, but python is still running")
+
+# wait until tapped
+await frame.sleep()
+print("Frame is sleeping, but python is still running")
+```
+
+</details>
+
+### Stay Awake
+
+```python
+async frame.stay_awake(value: bool)
+```
+
+Prevents Frame from going to sleep while it's docked onto the charging cradle.  This can help during development where continuous power is needed, however may degrade the display or cause burn-in if used for extended periods of time.
+
+There is no way to read the current value, only to set a new value.
+
+* `value` *(boolean)*: `True` for Frame to stay awake while charging, `False` to reset to normal operation so Frame turns off while charging.
+
+<details markdown="block">
+<summary>Python</summary>
+
+#### Python
+{: .no_toc }
+```python
+async def stay_awake(self, value: bool)
+```
+
+Example:
+
+```python
+await frame.stay_awake(True)
+# don't forget to turn this back off or you may damage your Frame
+```
+
+</details>
+
+### Send Break Signal
+
+```python
+async frame.bluetooth.send_break_signal(show_me: boolean = False)
+```
+
+Sends a break signal to the device which will break any currently executing Lua script.
+
+* `show_me` *(boolean)*: For debugging.  If `True`, the exact bytes sent to the device will be printed.
```

Comment:
Note to self: I removed `show_me` in favor of debugging being a stateful property, so I need to update this in the docs as well